src/containers/barray.mli

```diff
@@ -252,100 +216,71 @@ val bsearch : ?cmp:('a -> 'a -> int) -> 'a -> 'a t ->
 
-    @raise Invalid_argument if the array is found to be unsorted w.r.t [cmp]
-    @since 0.13 *)
-
-val for_all : ('a -> bool) -> 'a t -> bool
+    @raise Invalid_argument if the array is found to be unsorted w.r.t [cmp] *)
 
-val for_all2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
+val forAll2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
 (** Forall on pairs of arrays.
     @raise Invalid_argument if they have distinct lengths
-    allow different types @since 0.20 *)
-
-val exists : ('a -> bool) -> 'a t -> bool
+    allow different types *)
 
 val exists2 : ('a -> 'b -> bool) -> 'a t -> 'b t -> bool
 (** Exists on pairs of arrays.
     @raise Invalid_argument if they have distinct lengths
-    allow different types @since 0.20 *)
+    allow different types *)
 
 val fold2 : ('acc -> 'a -> 'b -> 'acc) -> 'acc -> 'a t -> 'b t -> 'acc
 (** Fold on two arrays stepwise.
-    @raise Invalid_argument if they have distinct lengths
-    @since 0.20 *)
+    @raise Invalid_argument if they have distinct lengths *)
 
 val iter2 : ('a -> 'b -> unit) -> 'a t -> 'b t -> unit
 (** Iterate on two arrays stepwise.
-    @raise Invalid_argument if they have distinct lengths
-    @since 0.20 *)
+    @raise Invalid_argument if they have distinct lengths *)
 
 val shuffle : 'a t -> unit
 (** Shuffle randomly the array, in place *)
 
-val shuffle_with : Random.State.t -> 'a t -> unit
+val shuffleWith : Random.State.t -> 'a t -> unit
 (** Like shuffle but using a specialized random state *)
 
-val random_choose : 'a t -> 'a random_gen
-(** Choose an element randomly.
-    @raise Not_found if the array/slice is empty *)
-
-val to_seq : 'a t -> 'a sequence
-val to_gen : 'a t -> 'a gen
-val to_klist : 'a t -> 'a klist
-
-(** {2 IO} *)
-
-val pp: ?sep:string -> 'a printer -> 'a t printer
-(** Print an array of items with printing function *)
-
-val pp_i: ?sep:string -> (int -> 'a printer) -> 'a t printer
-(** Print an array, giving the printing function both index and item *)
+val toIterator: 'a t -> 'a Iterator.t
+val fromIterator: 'a Iterator.t -> 'a t
 
 val map : ('a -> 'b) -> 'a t -> 'b t
 
 val map2 : ('a -> 'b -> 'c) -> 'a t -> 'b t -> 'c t
 (** Map on two arrays stepwise.
-      @raise Invalid_argument if they have distinct lengths
-      @since 0.20 *)
+      @raise Invalid_argument if they have distinct lengths *)
 
 val rev : 'a t -> 'a t
-(** Copy + reverse in place
-    @since 0.20 *)
+(** Copy + reverse in place *)
 
 val filter : ('a -> bool) -> 'a t -> 'a t
 (** Filter elements out of the array. Only the elements satisfying
     the given predicate will be kept. *)
 
-val filter_map : ('a -> 'b option) -> 'a t -> 'b t
+val filterMap : ('a -> 'b option) -> 'a t -> 'b t
 (** Map each element into another value, or discard it *)
 
-val flat_map : ('a -> 'b t) -> 'a t -> 'b array
+val flatMap : ('a -> 'b t) -> 'a t -> 'b array
 (** Transform each element into an array, then flatten *)
 
 val (>>=) : 'a t -> ('a -> 'b t) -> 'b t
```

Comment:
yep, random symbols are pretty hostile to newcomers, and they can easily be defined in your own projects or in an auxillary library.